### PR TITLE
Unified the formatting and added rail road diagrams

### PR DESCRIPTION
**Pull request by mikucionisaau**
- [x] reformatted most of the example code to be consistent
- [x] added [rail road diagrams](https://www.bottlecaps.de/rr/ui) for syntax (grammar.txt contains the massaged syntax)
- [x] fixed some spelling issues and other minor updates

### CODE REVIEW

File: content/language-reference/requirements-specification/_index.md [hunk 1]
```diff
@@ -120,6 +130,18 @@ Subjection ::=
 `control: A[] safe`
 : compute a strategy where `safe` state predicate is always true no matter what the oponent/environment chooses to do. The strategy is *permissive* in a sense that for a given state the strategy may propose multiple actions for the player/controller. Such permissive strategy can be thought of as a union of all strategies satisfying the predicate, therefore it does not have any notion of progress and may include infinite loops.
 
+`E<> control: A[] safe`
+: computes a collaborative strategy, i.e. finds a reachable state from where a safe strategy is found: `safe` state predicate is maintained to always remain true no matter what the oponent/environment chooses to do.
```
Comment (thorulf4) [inline content/language-reference/requirements-specification/_index.md:134]: This description didn't really explain the query to me, i've tried to change the sentence to highlight what i experienced/suprised me while testing this query type
```suggestion
: Checks if opponent and player can work together and reach one of the winning states of a given control strategy, note that the state predicate `safe` must always be true before reaching a winning state and is guaranteed to remain true after reaching winning state.

File: content/language-reference/requirements-specification/_index.md [hunk 2]
```diff
@@ -120,6 +130,18 @@ Subjection ::=
...
+
+`{ x<=5, Gate.Idle, Gate.Occ }: A[] safe`
+: computes a safe strategy by observing only the state of the specified predicate expressions.
```
Comment (thorulf4) [inline content/language-reference/requirements-specification/_index.md:137]:
```suggestion
: computes a safety strategy where the player can only observe the state of the specified predicate expressions.
```

File: content/language-reference/requirements-specification/_index.md [hunk 3]
```diff
@@ -120,6 +130,18 @@ Subjection ::=
 `control: A[] safe`
 : compute a strategy where `safe` state predicate is always true no matter what the oponent/environment chooses to do. The strategy is *permissive* in a sense that for a given state the strategy may propose multiple actions for the player/controller. Such permissive strategy can be thought of as a union of all strategies satisfying the predicate, therefore it does not have any notion of progress and may include infinite loops.
 
+`E<> control: A[] safe`
+: computes a collaborative strategy, i.e. finds a reachable state from where a safe strategy is found: `safe` state predicate is maintained to always remain true no matter what the oponent/environment chooses to do.
+
+`{ x<=5, Gate.Idle, Gate.Occ }: A[] safe`
```
Comment (thorulf4) [inline content/language-reference/requirements-specification/_index.md:136]: This line doesn't match the CFG given above.
By looking at the PartialControlSpecifier I expect the syntax:
`{ x<=5, Gate.Idle, Gate.Occ } control: A[] safe`

File: content/language-reference/requirements-specification/_index.md [hunk 4]
```diff
@@ -120,6 +130,18 @@ Subjection ::=
...
+: computes a safe strategy by observing only the state of the specified predicate expressions.
+
+`A[ safe U goal ]`
+: computes a safe strategy but only up until the `goal` is reached.
```
Comment (thorulf4) [inline content/language-reference/requirements-specification/_index.md:140]:
```suggestion
: computes a safety strategy but only up until the `goal` is reached.
```

File: content/language-reference/requirements-specification/_index.md [hunk 5]
```diff
@@ -120,6 +130,18 @@ Subjection ::=
...
+
+`A[ safe W goal ]`
+: (weakly until the `goal`) either a safe strategy is found or a safe strategy holds until the goal is reached.
```
Comment (thorulf4) [inline content/language-reference/requirements-specification/_index.md:143]:
```suggestion
: (weakly until the `goal`) either a safety strategy is found or a safety strategy holds until the goal is reached.
```

File: content/language-reference/requirements-specification/_index.md [hunk 6]
```diff
@@ -120,6 +130,18 @@ Subjection ::=
 `control: A[] safe`
 : compute a strategy where `safe` state predicate is always true no matter what the oponent/environment chooses to do. The strategy is *permissive* in a sense that for a given state the strategy may propose multiple actions for the player/controller. Such permissive strategy can be thought of as a union of all strategies satisfying the predicate, therefore it does not have any notion of progress and may include infinite loops.
```
Comment (thorulf4) [inline content/language-reference/requirements-specification/_index.md:131]:
```suggestion
: compute a strategy where `safe` state predicate is always true no matter what the opponent/environment chooses to do. The strategy is *permissive* in a sense that for a given state the strategy may propose multiple actions for the player/controller. Such permissive strategy can be thought of as a union of all strategies satisfying the predicate, therefore it does not have any notion of progress and may include infinite loops.
```

File: content/language-reference/system-description/declarations/types.md [hunk 7]
```diff
@@ -17,64 +17,65 @@ ArrayDecl     ::= '[' [Expression] ']'
...
-Integers, booleans, doubles strings, and arrays and records over integers and booleans can be marked constant by prefixing the type with the keyword <tt>const</tt>.
+Integers, booleans, doubles strings, and arrays and records over integers and booleans can be marked constant by prefixing the type with the keyword `const`.
 
 ## Meta variables
 
-Integers, booleans, doubles, and arrays and records over integers and booleans can be marked as meta variables by prefixing the type with the keyword <tt>meta</tt>.
+Integers, booleans, doubles, and arrays and records over integers and booleans can be marked as meta variables by prefixing the type with the keyword `meta`.
 
-Meta variables are stored in the state vector, but are sematically not considered part of the state. I.e. two states that only differ in meta variables are considered to be equal.
+Meta variables are stored outside of the state vector and are sematically not considered part of the state. I.e. two states that only differ in meta variables are considered to be equal.
```
Comment (thorulf4) [inline content/language-reference/system-description/declarations/types.md:43]:
```suggestion
Meta variables are stored outside of the state vector and are semantically not considered part of the state. I.e. two states that only differ in meta variables are considered to be equal.
```

File: content/language-reference/system-description/templates/edges.md [hunk 8]
```diff
@@ -5,27 +5,20 @@ weight: 20
...
+Guards
+: An edge is enabled in a state if and only if the guard evaluates to true.
 
-<dd>Selections non-deterministically bind a given identifier to a value in a given range. The other three labels of an edge are within the scope of this binding.</dd>
+Synchronisation
+: Processes can synchronize over channels. Edges labelled with complementary actions over a common channel synchronise.
 
-<dt>Guards</dt>
+Updates
+: When executed, the update expression of the edge is evaluated. The side effect of this expression changes the state of the system.
```
Comment (thorulf4) [inline content/language-reference/system-description/templates/edges.md:18]:
```suggestion
: When an edge is traversed, the update expression of the edge is evaluated. The side effect of this expression changes the state of the system.
```

**Review by thorulf4 — CHANGES_REQUESTED** [review]

File: content/language-reference/requirements-specification/_index.md (hunk 3 above)
Comment (mikucionisaau) [reply content/language-reference/requirements-specification/_index.md:136]: Good catch!

Comment (thorulf4) [inline content/language-reference/requirements-specification/_index.md:136]: This query type does not work in Uppaal 5 it throws a segmentation fault when trying

**Pull request merged**